src/components/Modals/NewCourse/CourseSelector.vue

```diff
@@ -80,7 +57,32 @@ export default defineComponent({
   },
   computed: {
     matches(): readonly CornellCourseRosterCourse[] {
-      return getMatchingCourses(this.searchText.toUpperCase(), this.courseFilter);
+      if (!this.searchText || this.searchText.length < 2) return [];
+      const options = {
+        keys: [
+          { name: 'courseCode', weight: 2 },
+          { name: 'subject', weight: 3 },
+          { name: 'fullCourseString', weight: 2.5 },
+          { name: 'catalogNbr', weight: 1.5 },
+          { name: 'titleLong', weight: 1 },
+        ],
+      };
+
+      const fuse = new Fuse(this.searchableCourses, options);
+      const result: readonly SearchableCourse[] = fuse
+        .search(this.searchText, { limit: 10 })
+        .map(elem => elem.item);
+      return result;
+    },
+    searchableCourses(): readonly SearchableCourse[] {
+      const courses: readonly SearchableCourse[] =
+        this.courseFilter != null ? fullCoursesArray.filter(this.courseFilter) : fullCoursesArray;
+      courses.map((course: SearchableCourse) => {
+        course.courseCode = `${course.subject} ${course.catalogNbr}`;
+        course.fullCourseString = `${course.subject} ${course.catalogNbr} ${course.titleLong}`;
+        return course;
+      });
```

Comment:
Could we define a const like `fullSearchableCoursesArray` outside of computed props such that
```
fullSearchableCoursesArray = fullCoursesArray.map((course: SearchableCourse) => {
        course.courseCode = `${course.subject} ${course.catalogNbr}`;
        course.fullCourseString = `${course.subject} ${course.catalogNbr} ${course.titleLong}`;
        return course;
      });
```
and then the computed prop`searchableCourses()` would be `this.courseFilter != null ? fullSearchableCoursesArray.filter(this.courseFilter) : fullSearchableCoursesArray;`.

Then, we wouldn't have to add the `courseCode` and `fullCourseString` field every time we search for courses because it would precomputed. Hopefully this will speed things up a bit!